### PR TITLE
Implement docker environment

### PR DESCRIPTION
**Pull request by jamescurtin**
Submitting this PR in response to issue https://github.com/sloria/cookiecutter-flask/issues/65, as it doesn't seem like PR #272 has activity in > 1 year.

# What is added
* Support for running the cookiecutter app in Docker
    * User specifies desired Python and Node version during cookiecutter setup
    * Separate production and development docker-compose services
        * Production service leverages gunicorn and supervisord 
* Misc. Fixes & additions
    *  `app_name` from the `cookiecutter.json` must be a valid Python module name, but there is no validation. The default is now the Python-friendly version of the `project_name` to reduce possibility for user error.
    * PDB++ debugger

### CODE REVIEW

**Comment by jamescurtin** [comment]
👋 @sloria Apologies for the ping; is this PR something you're open to reviewing? Thank you!

**Comment by philiphoyos** [comment]
I can help test, however, I don't know who to make cookiecutter use this commit?

**Comment by sloria** [comment]
Once the .gitignore change is reverted and this is updated with the latest master (rebase preferred, but merge is fine, too), this is good to merge.

File: cookiecutter.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
 	"email": "sloria1@gmail.com",
 	"github_username": "sloria",
 	"project_name": "My Flask App",
-	"app_name": "myflaskapp",
+	"app_name": "{{cookiecutter.project_name.lower().replace('-', '_').replace(' ', '_')}}",
```
[the remark this answers is not in the document]
Comment (jamescurtin) [reply cookiecutter.json]: 👍 Thanks!

File: .gitignore [hunk 2]
```diff
@@ -44,5 +44,11 @@ README.html
...
 output/
 
+# Mypy
+.mypy_cache/
+
 myflaskapp/
 bower_components
+
+# MacOS
+.DS_Store
```
[the remark this answers is not in the document]
Comment (jamescurtin) [reply .gitignore]: 👍

**Comment by jamescurtin** [comment]
Great--thanks for the review! I've made the requested changes & rebased to master. 

I'd be excited to become a maintainer; expect a few more PRs in the next few weeks!

**Comment by hassanc** [comment]
@jamescurtin this works great... but I can't figure out the URL to connect to once I have `docker-compose up flask-dev` running.

I checked `localhost:5000`, `ifconfig` to find the my Ubunutu VM IP, `0.0.0.0:5000` and used docker inspect to view the IP address. None of them seemed to be working.

What IP address should I be using?

**Comment by jamescurtin** [comment]
Hi @hassanc, I've moved your question over to an issue so that it may be better tracked. Thanks!

**Pull request merged**